app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -608,30 +615,34 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
 
     // Check whether palm of left hand is facing the head, to show navigate-back button
     bool leftPalmFacesHead = false;
-    if (mHandeness == OpenXRHandFlags::Left) {
-#if defined(OCULUSVR)
-        if (IsHandJointPositionValid(XR_HAND_JOINT_PALM_EXT))
-            leftPalmFacesHead = !mHasAimState;
-#else
-        XrHandJointEXT palmJointIndex = XR_HAND_JOINT_PALM_EXT;
-#if defined(PICOXR)
-        // Pico runtime doesn't provide palm joint info, so we use the wrist instead.
-        palmJointIndex = XR_HAND_JOINT_WRIST_EXT;
+    bool systemTakesOverWhenHandsFacingHead = false;
+#if defined(OCULUS)
+    systemTakesOverWhenHandsFacingHead = true;
 #endif
-        if (IsHandJointPositionValid(palmJointIndex)) {
-            vrb::Matrix palmMatrix = jointTransforms[palmJointIndex];
-            // For the hand we take the Y axis because that corresponds to head's Z axis when
-            // the hand is in upright position facing head (the gesture we want to detect).
+    if (IsHandJointPositionValid(HAND_JOINT_FOR_AIM)) {
+        bool palmFacesHead = false;
+        if (mSupportsFBHandTrackingAim) {
+            // With the FB aim extension we stop getting aim state precisely when hands face head.
+            palmFacesHead = !mHasAimState;
+        } else {
+            if (systemTakesOverWhenHandsFacingHead) {
+                palmFacesHead = !mHasAimState;
+            } else {
+                vrb::Matrix palmMatrix = jointTransforms[HAND_JOINT_FOR_AIM];
+                // For the hand we take the Y axis because that corresponds to head's Z axis when
+                // the hand is in upright position facing head (the gesture we want to detect).
 #ifdef PICOXR
-            // Y-Z Axis are inverted in Pico
-            auto vectorPalm = palmMatrix.MultiplyDirection({0, 0, -1});
+                // Y-Z Axis are inverted in Pico
+                auto vectorPalm = palmMatrix.MultiplyDirection({0, 0, -1});
 #else
-            auto vectorPalm = palmMatrix.MultiplyDirection({0, 1, 0});
+                auto vectorPalm = palmMatrix.MultiplyDirection({0, 1, 0});
 #endif
-            auto vectorHead = head.MultiplyDirection({0, 0, -1});
-            leftPalmFacesHead = vectorPalm.Dot(vectorHead) > kPalmHeadThreshold;
+                auto vectorHead = head.MultiplyDirection({0, 0, -1});
+                palmFacesHead = vectorPalm.Dot(vectorHead) > kPalmHeadThreshold;
+                mHasAimState = mHasAimState && !palmFacesHead;
+            }
         }
-#endif
+        leftPalmFacesHead = mHandeness == Left && leftPalmFacesHead;
```

Comment:
Maybe grouping parenthesis around the first condition?

Reply:
equality comparisons  have precedence over logical operations (even bitwise) so it should work. If it's a matter just of taste well I think this is more clean, but I'll change it if you strongly feel it's preferrable.